symengine/polys/uintpoly.cpp

```diff
@@ -130,4 +139,60 @@ bool UIntPoly::is_pow() const
     return false;
 }
 
+RCP<const UIntPoly> pow_upoly(const UIntPoly &a, unsigned int p)
+{
+    auto tmp = a.get_poly();
+    UIntDict res(1);
+
+    while (p > 1) {
```

Comment:
Initially I had done what you have suggested. When `p = 1`, there will be an extra unnecessary calculation `tmp = tmp * tmp`. Otherwise it's the same.